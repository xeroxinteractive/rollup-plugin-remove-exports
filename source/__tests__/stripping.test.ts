import stripExports from '../index';
import { rollup } from 'rollup';
import globby from 'globby';
import path from 'path';

test.each(
  globby.sync(path.join(__dirname, '__specs__')).map((pathname) => {
    const { name } = path.parse(pathname);
    return [name, pathname];
  })
)('%s', async (name, pathname) => {
  const bundle = await rollup({
    input: pathname,
    treeshake: false,
    plugins: [stripExports()],
  });
  const { output } = await bundle.generate({ format: 'es' });

  expect(output[0].code).toMatchSnapshot();
});
