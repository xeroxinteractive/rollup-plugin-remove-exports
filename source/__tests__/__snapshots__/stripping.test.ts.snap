// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all-from 1`] = `
"
"
`;

exports[`as 1`] = `
"const test = 'string';
"
`;

exports[`class 1`] = `
"class Cat {
  constructor() {}
}
"
`;

exports[`decleration 1`] = `
"const test = 'string';
"
`;

exports[`default-as 1`] = `
"
"
`;

exports[`default-class 1`] = `
"class Dog {
  constructor() {}
}
"
`;

exports[`default-class-instance 1`] = `
"class Iguana {
  constructor() {}
}

new Iguana();
"
`;

exports[`default-const 1`] = `
"const value = 'string';
"
`;

exports[`default-function 1`] = `
"function name(x) {
  return x;
}
"
`;

exports[`default-literal 1`] = `
"
"
`;

exports[`default-named 1`] = `
"const test = 'string';
const test1 = 5;
const test3 = false;
const test4 = undefined;
const test5 = new Symbol('duck');

foo => {
  console.log(foo + 'bar');
};
"
`;

exports[`multiple-named 1`] = `
"const test = 'string';

function test1() {
  return 'string';
}

let inline = () => false;
"
`;
